plugins/copy-by-storage/src/index.js

```diff
@@ -0,0 +1,206 @@
+      Blockly.utils.KeyCodes.C, [Blockly.utils.KeyCodes.CTRL]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(ctrlC, copyShortcut.name);
+
+  const altC =
+    Blockly.ShortcutRegistry.registry.createSerializedKey(
+        Blockly.utils.KeyCodes.C, [Blockly.utils.KeyCodes.ALT]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(altC, copyShortcut.name);
+
+  const metaC = Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.C, [Blockly.utils.KeyCodes.META]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(metaC, copyShortcut.name);
+};
+Blockly.ShortcutItems.blockCutToStorage = function() {
+/** @type {!Blockly.ShortcutRegistry.KeyboardShortcut} */
+  const cutShortcut = {
+    name: 'cut',
+    preconditionFn: function(workspace) {
+      return !workspace.options.readOnly &&
+        !Blockly.Gesture.inProgress() &&
+        Blockly.selected &&
+        Blockly.selected.isDeletable() &&
+        Blockly.selected.isMovable() &&
+        !Blockly.selected.workspace.isFlyout;
+    },
+    callback: function(workspace, e) {
+      // Prevent the default copy behavior, which may beep or otherwise indicate
+      // an error due to the lack of a selection.
+      e.preventDefault();
+      const blockDom = Blockly.Xml.blockToDomWithXY(Blockly.selected);
+      Blockly.Xml.deleteNext(blockDom);
+      const blockText = Blockly.Xml.domToText(blockDom);
+      Blockly.copyByStorage.__Storage.setItem('blocklyStash', blockText);
+      Blockly.deleteBlock(Blockly.selected);
+      return true;
+    },
+  };
+  Blockly.ShortcutRegistry.registry.register(cutShortcut);
+
+  const ctrlX = Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.X, [Blockly.utils.KeyCodes.CTRL]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(ctrlX, cutShortcut.name);
+
+  const altX = Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.X, [Blockly.utils.KeyCodes.ALT]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(altX, cutShortcut.name);
+
+  const metaX = Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.X, [Blockly.utils.KeyCodes.META]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(metaX, cutShortcut.name);
+};
+Blockly.ShortcutItems.blockPasteFromStorage = function() {
+/** @type {!Blockly.ShortcutRegistry.KeyboardShortcut} */
+  const pasteShortcut = {
+    name: 'paste',
+    preconditionFn: function(workspace) {
+      if (!Blockly.copyByStorage.__Storage.getItem('blocklyStash')) {
+        return 'disabled';
+      }
+      return !workspace.options.readOnly && !Blockly.Gesture.inProgress();
+    },
+    callback: function(workspace, e) {
+      // Prevent the default copy behavior, which may beep or otherwise indicate
+      // an error due to the lack of a selection.
+      e.preventDefault();
+      const blockText = Blockly.copyByStorage.__Storage.getItem('blocklyStash');
```

Comment:
Instead of creating `Blockly.copyByStorage.__Storage` I think we can just use localStorage here.